app/models/user.rb

```diff
@@ -7,4 +7,9 @@ class User < ActiveRecord::Base
 
   has_many :bookmarks
   has_many :topics
+  has_many :likes, dependent: :destroy
+
+  def liked(bookmark)
```

Comment:
I'd change this to a declarative method: `liked?(bookmark)` and return a try or false value. It'd make it easier to use in the view.